src/exceptions.md

```diff
@@ -1,33 +1,36 @@
 # Exception handling
 
-During the "Memory layout" section we decided to start out simple and leave out handling of
-exceptions. In this section we'll add support for handling exceptions; this serves as an example of
+During the "Memory layout" section, we decided to start out simple and leave out handling of
+exceptions. In this section, we'll add support for handling them; this serves as an example of
 how to achieve compile time overridable behavior in stable Rust (i.e. without relying on the
 unstable `#[linkage = "weak"]` attribute, which makes a symbol weak).
 
 ## Background information
 
 In a nutshell, *exceptions* are a mechanism the Cortex-M and other architectures provide to let
-applications respond to asynchronous, usually external, events. The exception mechanism works like
-this: when the processor receives a signal or event associated to a type of exception it suspends
+applications respond to asynchronous, usually external, events. The most prominent type of exception,
+that most people will know, is the classical (hardware) interrupt.
+
+The Cortex-M exception mechanism works like this:
+When the processor receives a signal or event associated to a type of exception, it suspends
 the execution of the current subroutine (by stashing the state in the call stack) and then proceeds
 to execute the corresponding exception handler, another subroutine, in a new stack frame. After
-finishing the execution of the exception handler (i.e. returning from it) the processor resumes the
+finishing the execution of the exception handler (i.e. returning from it), the processor resumes the
 execution of the suspended subroutine.
 
 The processor uses the vector table to decide what handler to execute. Each entry in the table
 contains a pointer to a handler, and each entry corresponds to a different exception type. For
 example, the second entry is the reset handler, the third entry is the NMI (Non Maskable Interrupt)
 handler, and so on.
 
-As mentioned before the processor expects the vector table to be at some specific location and each
-entry in it can potentially be used by the processor at runtime so they must always contain valid
-values. Furthermore, we want the `rt` crate to be flexible so the end user can customize the
+As mentioned before, the processor expects the vector table to be at some specific location in memory,
+and each entry in it can potentially be used by the processor at runtime. Hence, the entries must always
+contain valid values. Furthermore, we want the `rt` crate to be flexible so the end user can customize the
 behavior of each exception handler. Finally, the vector table is read only memory, or rather in not
```

Comment:
nit: not part of your changes but this would read better as "the vector table *resides in* read only memory".